gpu_tests/test_hf_compatibility.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
+import os
+from transformers import GPT2Tokenizer
+from metaseq import checkpoint_utils, tasks, utils
+from transformers import OPTForCausalLM
+from packaging import version
+import torch
+import unittest
+import torch.nn.functional as F
+from metaseq.scripts.convert_to_singleton import create_generation_config_with_defaults
+from metaseq.distributed import utils as dist_utils
+from metaseq.distributed import fsdp_enable_wrap, fsdp_wrap
+from metaseq.dataclass.configs import MetaseqConfig
+
+
+prompts = [
+    "Today is a beautiful day and I want to",
+    "In the city of",
+    "Paris is the capital of France and",
+    "Computers and mobile phones have taken",
+]
+
+
+# forward passes
+def tensorize_input(tokenizer, prompts):
+    input_ids = tokenizer(prompts, return_tensors="pt").input_ids
+    input_ids = torch.cat([torch.tensor([[0]]), input_ids], dim=-1)
+    input_ids = input_ids
+    return input_ids
+
+
+def get_next_token(logits, tokenizer):
+    pred_next_token = torch.argmax(logits[0, -1], -1)
+    next_token = tokenizer.convert_ids_to_tokens([pred_next_token])
+    next_token = next_token[0].replace("Ġ", "")
+    return next_token
+
+
+def setup_vocab_and_merges(model_path):
+    vocab_file = os.path.join(model_path, "gpt2-vocab.json")
+    merges_file = os.path.join(model_path, "gpt2-merges.txt")
+    tokenizer = GPT2Tokenizer(vocab_file, merges_file)
+    tokenizer.save_pretrained(model_path)
+    return vocab_file, merges_file, tokenizer
+
+
+def load_mp_model_and_run_eval(cfg: MetaseqConfig, **kwargs):
```

Comment:
Should these things be moved to hubutils or similar?

Reply:
Moved the common functions to hub_utils.py

I've kept load_mp_model_and_run_eval in this file still, since the logic there is very specific to this test... And we are using common functions in the function body (like fsdp_enable_wrap, or checkpoint_utils.load_model_ensemble_and_task).